### PR TITLE
Added a function to check for package manager locks

### PR DESCRIPTION
**Pull request by reaperzn**

### CODE REVIEW

**Comment by reaperzn** [comment]
This is a possible fix for #101

File: auter.yumdnfModule [hunk 1]
```diff
@@ -9,6 +9,29 @@ if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
...
+  [[ "${PACKAGEMANAGERLOCKRETRIES}" -lt 1 ]] && PACKAGEMANAGERLOCKRETRIES=5
+
+  [[ "${PACKAGEMANAGERLOCKWAITTIME}" ]] || PACKAGEMANAGERLOCKWAITTIME=60
+  [[ "${PACKAGEMANAGERLOCKWAITTIME}" -lt 1 ]] && PACKAGEMANAGERLOCKWAITTIME=60
+
+  if [[ -f /var/run/yum.pid ]]; then
+    for LOCKCHECKATTEMPT in $(seq 1 $[${PACKAGEMANAGERLOCKRETRIES}+1]); do
+      # If this has reached the retry limit configured then abort
+      if [[ ${LOCKCHECKATTEMPT} -eq $[${PACKAGEMANAGERLOCKRETRIES}+1] ]]; then
+        logit "Final attemt to wait for yum to release lock file failed. Aborting auter run"
```
Comment (Pingue) [inline auter.yumdnfModule]: Spelling

**Review by Pingue — CHANGES_REQUESTED** [review]

**Review by Pingue — APPROVED** [review]

File: auter.yumdnfModule [hunk 2]
```diff
@@ -9,6 +9,29 @@ if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
...
+  [[ "${PACKAGEMANAGERLOCKRETRIES}" -lt 1 ]] && PACKAGEMANAGERLOCKRETRIES=5
+
+  [[ "${PACKAGEMANAGERLOCKWAITTIME}" ]] || PACKAGEMANAGERLOCKWAITTIME=60
+  [[ "${PACKAGEMANAGERLOCKWAITTIME}" -lt 1 ]] && PACKAGEMANAGERLOCKWAITTIME=60
+
+  if [[ -f /var/run/yum.pid ]]; then
+    for LOCKCHECKATTEMPT in $(seq 1 $[${PACKAGEMANAGERLOCKRETRIES}+1]); do
+      # If this has reached the retry limit configured then abort
+      if [[ ${LOCKCHECKATTEMPT} -eq $[${PACKAGEMANAGERLOCKRETRIES}+1] ]]; then
+        logit "Final attempt to wait for yum to release lock file failed. Aborting auter run"
```
Comment (Pingue) [inline auter.yumdnfModule]: Should this start with "INFO:" for consistency?

File: auter.yumdnfModule [hunk 3]
```diff
@@ -9,6 +9,29 @@ if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then
...
+        quit 3
+      fi
+
+      logit "/var/run/yum.pid exists. Waiting ${PACKAGEMANAGERLOCKWAITTIME} seconds for yum lock to be released: Attempt ${LOCKCHECKATTEMPT} of ${PACKAGEMANAGERLOCKRETRIES}"
+      sleep ${PACKAGEMANAGERLOCKWAITTIME}
+      [[ ! -f /var/run/yum.pid ]] && logit "yum lock has been released" && break
```
Comment (Pingue) [inline auter.yumdnfModule]: As above - should this start with INFO?

**Review by Pingue — CHANGES_REQUESTED** [review]
I've tested on CentOS 6.9, without problems - I'll do some more testing on other OS.

**Review by Pingue — APPROVED** [review]

**Comment by Pingue** [comment]
Tested and working on RHEL6 and RHEL7. Apt has the framework, but is not implemented, same for DNF, however this doesn't make any changes to these systems. Merging this for now, DNF and apt will need to be added later.

**Pull request merged**